docs/source/tutorial_external_solver.rst

```diff
@@ -0,0 +1,383 @@
+of |VectorSpace| are default implemented in :class:`~pymor.vectorarrays.list.ListVectorSpace`.
+
+.. jupyter-execute::
+
+  class WrappedVectorSpace(ListVectorSpace):
+
+      def __init__(self, dim):
+          self.dim = dim
+
+      def zero_vector(self):
+          return WrappedVector(Vector(self.dim, 0))
+
+      def make_vector(self, obj):
+          # obj is a `model.Vector` instance
+          return WrappedVector(obj)
+
+      def __eq__(self, other):
+          return type(other) is WrappedVectorSpace and self.dim == other.dim
+
+
+Wrapping the `model.DiffusionOperator` is straight forward as well. We just need to attach
+suitable |VectorSpaces| to the class and implement the application of the operator on a |VectorArray|
+as a sequence of applications on single vectors.
+
+.. jupyter-execute::
+
+  class WrappedDiffusionOperator(Operator):
+      def __init__(self, op):
+          assert isinstance(op, DiffusionOperator)
+          self.op = op
+          self.source = WrappedVectorSpace(op.dim_source)
+          self.range = WrappedVectorSpace(op.dim_range)
+          self.linear = True
+
+      @classmethod
+      def create(cls, n, left, right):
+          return cls(DiffusionOperator(n, left, right))
+
+      def apply(self, U, mu=None):
+          assert U in self.source
+
+          def apply_one_vector(u):
+              v = Vector(self.range.dim, 0)
+              self.op.apply(u._impl, v)
+              return v
+
+          return self.range.make_array([apply_one_vector(u) for u in U._list])
+
+
+Putting it all together
+-----------------------
+
+As a demonstration we will use our (stationary) toy diffusion solver in a transient example with timestepping
+provided by pyMOR. For details on the discretization setup see :doc:`tutorial01`. First up we implement
+a `discretize` function that uses the `WrappedDiffusionOperator` and `WrappedVectorSpace` to assemble an
+|InstationaryModel|.
+
+.. jupyter-execute::
+
+  from pymor.algorithms.pod import pod
+  from pymor.algorithms.timestepping import ExplicitEulerTimeStepper
+  from pymor.discretizers.builtin.gui.visualizers import OnedVisualizer
+  from pymor.models.basic import InstationaryModel
+  from pymor.discretizers.builtin import OnedGrid
+  from pymor.operators.constructions import VectorOperator, LincombOperator
+  from pymor.parameters.functionals import ProjectionParameterFunctional
+  from pymor.reductors.basic import InstationaryRBReductor
+
+
+  def discretize(n, nt, blocks):
+      h = 1. / blocks
+      ops = [WrappedDiffusionOperator.create(n, h * i, h * (i + 1)) for i in range(blocks)]
+      pfs = [ProjectionParameterFunctional('diffusion_coefficients', blocks, i) for i in range(blocks)]
+      operator = LincombOperator(ops, pfs)
+
+      initial_data = operator.source.zeros()
+
+      # use to_numpy method of WrappedVector to setup rhs
+      # note that we cannot use the data property of ListVectorArray,
```

Comment:
The `data` property is deprecated.

Reply:
Hah, caught in the first, but not the second line :roll_eyes: